lib/northbound.c

```diff
@@ -1664,79 +1654,18 @@ int nb_notification_send(const char *xpath, struct list *arguments)
 	return ret;
 }
 
-/* Running configuration user pointers management. */
-struct nb_config_entry {
-	char xpath[XPATH_MAXLEN];
-	void *entry;
-};
-
-static bool running_config_entry_cmp(const void *value1, const void *value2)
-{
-	const struct nb_config_entry *c1 = value1;
-	const struct nb_config_entry *c2 = value2;
-
-	return strmatch(c1->xpath, c2->xpath);
-}
-
-static unsigned int running_config_entry_key_make(const void *value)
-{
-	return string_hash_make(value);
-}
-
-static void *running_config_entry_alloc(void *p)
-{
-	struct nb_config_entry *new, *key = p;
-
-	new = XCALLOC(MTYPE_NB_CONFIG_ENTRY, sizeof(*new));
-	strlcpy(new->xpath, key->xpath, sizeof(new->xpath));
-
-	return new;
-}
-
-static void running_config_entry_free(void *arg)
-{
-	XFREE(MTYPE_NB_CONFIG_ENTRY, arg);
-}
-
 void nb_running_set_entry(const struct lyd_node *dnode, void *entry)
 {
-	struct nb_config_entry *config, s;
-
-	yang_dnode_get_path(dnode, s.xpath, sizeof(s.xpath));
-	config = hash_get(running_config_entries, &s,
-			  running_config_entry_alloc);
-	config->entry = entry;
-}
-
-static void *nb_running_unset_entry_helper(const struct lyd_node *dnode)
-{
-	struct nb_config_entry *config, s;
-	struct lyd_node *child;
-	void *entry = NULL;
-
-	yang_dnode_get_path(dnode, s.xpath, sizeof(s.xpath));
-	config = hash_release(running_config_entries, &s);
-	if (config) {
-		entry = config->entry;
-		running_config_entry_free(config);
-	}
-
-	/* Unset user pointers from the child nodes. */
-	if (CHECK_FLAG(dnode->schema->nodetype, LYS_LIST | LYS_CONTAINER)) {
-		LY_TREE_FOR (dnode->child, child) {
-			(void)nb_running_unset_entry_helper(child);
-		}
-	}
-
-	return entry;
+	lyd_set_private(dnode, entry);
 }
 
 void *nb_running_unset_entry(const struct lyd_node *dnode)
 {
 	void *entry;
 
-	entry = nb_running_unset_entry_helper(dnode);
+	entry = dnode->priv;
 	assert(entry);
+	lyd_set_private(dnode, NULL);
```

Comment:
we can simply do entry = lyd_set_private(donode,NULL); as it returns old set previous value and that can be returned.

Reply:
Good point. I'll update the code accordingly.